cpp/velox/shuffle/VeloxShuffleWriter.cc

```diff
@@ -1115,7 +1115,9 @@ arrow::Status VeloxShuffleWriter::splitFixedWidthValueBuffer(const velox::RowVec
       const arrow::RecordBatch& rb, bool reuseBuffers) {
     auto payload = std::make_shared<arrow::ipc::IpcPayload>();
 #ifndef SKIPCOMPRESS
-    auto isTinyBatch = rb.num_rows() <= options_.batch_compress_threshold;
+    // Extract numRows from header column
```

Comment:
Can we add the numRows to `createArrowIpcPayload` argument? Header buffer code maybe changed and may not pick up here.

Reply:
Or check the rb.column_data(0)->buffers[2] size, make sure it is numRows